### PR TITLE
DisplayServer: Some fixes to X11, build system and Window class

### PR DESCRIPTION
**Pull request by kuruk-mm**
Includes and Fix #37002

CI repaired (Mac is failing, I'm not going to fix that)
X11 now is compiling, working ~(but with love)~, and no crashing your entire desktop gui

Known bugs:
- ~Main Window size changed when you close a subwindow~
- ~Popups are broken (reduz is working on this)~
- ~When you open a dropdown menu it crash (with bad parameter at XCreateWindow)~
- Popups are working, but when you move a Window, and you open a Popup, the Popup appear in the last position
- Dropdown menus appear a little above what they should (seems to be window top margin)

@reduz There are simple changes, if you want to merge, and continue working from this is welcome

There still have a work to do, just making this PR to sync our work

### CODE REVIEW

File: editor/project_manager.cpp
Comment (reduz) [inline editor/project_manager.cpp]: Probably don't need to fix this, as Window now properly computes the minimum size and enforces it.

**Review by reduz — CHANGES_REQUESTED** [review]

File: main/main_builders.py
Comment (reduz) [inline main/main_builders.py]: I think controller mappings are gone, so the function to make controller mappings should be gone from here too

**Review by reduz — CHANGES_REQUESTED** [review]

File: scene/gui/option_button.cpp
Comment (reduz) [inline scene/gui/option_button.cpp]: Not needed any longer, since minsize bug is fixed

**Review by reduz — CHANGES_REQUESTED** [review]

File: platform/linuxbsd/display_server_x11.cpp
Comment (reduz) [inline platform/linuxbsd/display_server_x11.cpp]: What is this for? not quite sure why the main window should not notify size changed

**Review by reduz — CHANGES_REQUESTED** [review]

File: platform/linuxbsd/display_server_x11.cpp
Comment (kuruk-mm) [reply platform/linuxbsd/display_server_x11.cpp]: Because when you close a window and remove it from the windows map, all _window_changed events were going to the MAIN_WINDOW and making a lot of weird things.

File: main/main_builders.py
Comment (kuruk-mm) [reply main/main_builders.py]: So you mean, you want me to delete everything about controller mapping? (code generator and controllers db txt)

And this:
https://github.com/godotengine/godot/blob/8874de089896cefcccd0a85b24d0b67ed86b6df4/core/input/input_filter.cpp#L1296-L1301
Because I followed the pipe, and I need to delete this too.

**Comment by kuruk-mm** [comment]
I make a new commit to make easier to review

Done, please before merge, let me know, because I want to squash the commits to have a clean git history

**Comment by kuruk-mm** [comment]
Ready to merge as you requested on IRC @reduz 

Know bug: Popups are working, but when you move a Window, and you open a Popup, the Popup appear in the last position

Please take care from this:
https://github.com/kuruk-mm/godot/blob/7b5e9788a8d4872292bf8f1f776831e67d42936e/platform/linuxbsd/display_server_x11.cpp#L2240-L2248

If you delete a window and an event was pending, MainWindow will receive it instead of the deleted window. Anyway I think this can be Ok because is Thread Safe, so I just remark to have a knowledge

File: main/main_builders.py
Comment (akien-mga) [inline main/main_builders.py]: For the reference if anyone wonders about this removal: this code was moved to `core/input/`.

File: platform/linuxbsd/display_server_x11.cpp
Comment (akien-mga) [inline platform/linuxbsd/display_server_x11.cpp]: Should this stay enabled, or was it only for temporary debugging?

Comment (kuruk-mm) [reply platform/linuxbsd/display_server_x11.cpp]: Is very useful, I use this for the knowing bugs.

**Review by akien-mga — APPROVED** [review]
Looks good to merge if @reduz approves.

**Comment by kuruk-mm** [comment]
Updated from @reduz last changes

Popups are working fine
Known bug:
- When you make the Node Inspector floating, and close, sometimes will crash and sometimes stop working

**Pull request merged**
